skopt/optimizer/optimizer.py

```diff
@@ -446,6 +450,31 @@ def tell(self, x, y, fit=True):
                 warnings.simplefilter("ignore")
                 est.fit(self.space.transform(self.Xi), self.yi)
 
+            # if we are using noisy expected improvement generate N variants
+            # of the observed data and compute a GP for each variant.
+            # The variable names reflect the notation used in the original 
+            # paper (Algorithm 1)
+            if self.acq_func == "noisyEI":
+                mu, sigma = est.predict(self.Xi, return_cov=True)
```

Comment:
It might also be a good idea to check if we achieve similar performances just by sampling randomly, you can use the `sample_y` function to sample from the posterior directly.